esmvaltool/cmorizers/obs/osi_common.py

```diff
@@ -0,0 +1,247 @@
+            'Only %s days available. Can not generate daily files',
+            cube.coord('time').shape[0]
+        )
+        return None
+    total_days = 366 if isleap(year) else 365
+    if cube.coord('time').shape[0] < total_days:
+        cubes = _add_nan_timesteps(cube, total_days)
+        cube = cubes.merge_cube()
+        cube.remove_coord('day_of_year')
+        del cubes
+    return cube
+
+
+def _add_nan_timesteps(cube, total_days):
+    add_day_of_year(cube, 'time')
+    cubes = CubeList(cube.slices_over('time'))
+    model_cube = cubes[0].copy()
+    model_cube.remove_coord('day_of_year')
+    for day_of_year in range(total_days):
+        day_constraint = iris.Constraint(
+            day_of_year=day_of_year + 1
+        )
+        if cubes.extract(day_constraint):
+            continue
+        nan_cube = _create_nan_cube(
+            model_cube, day_of_year, month=False
+        )
+        add_day_of_year(nan_cube, 'time')
+        cubes.append(nan_cube)
+    del model_cube
+    return cubes
+
+
+def _create_nan_cube(model_cube, num, month):
+    nan_cube = model_cube.copy(
+        np.ma.masked_all(model_cube.shape, dtype=model_cube.dtype)
+    )
+    time_coord = nan_cube.coord('time')
+    nan_cube.remove_coord(time_coord)
+    date = time_coord.cell(0).point
+    if month:
+        date = datetime(date.year, num, date.day)
+        bounds = (
+            datetime(date.year, num, 1),
+            datetime(date.year, num, monthrange(date.year, num)[1])
+        )
+    else:
+        date = datetime(date.year, 1, 1, 12) + timedelta(days=num)
+        bounds = (
+            datetime(date.year, 1, 1) + timedelta(days=num),
+            datetime(date.year, 1, 1, 23, 59) + timedelta(days=num)
+        )
+
+    date = time_coord.units.date2num(date)
+    bounds = (
+        time_coord.units.date2num(bounds[0]),
+        time_coord.units.date2num(bounds[1]),
+    )
+    nan_cube.add_aux_coord(AuxCoord(
+        [date],
+        standard_name=time_coord.standard_name,
+        var_name=time_coord.var_name,
+        long_name=time_coord.long_name,
+        units=time_coord.units,
+        attributes=time_coord.attributes,
+        bounds=[bounds],
+    ))
+    return nan_cube
+
+
+@vectorize(['float32(float32)'])
+def _correct_lons(lon):
+    if lon < 0:
+        return lon + 360
+    return lon
+
+
+def _unify_attributes(cubes):
+    tracking_ids = []
+    for cube in cubes:
+        # OSI-409 and OSI-450 do not have the same attributes
+        try:
+            tracking_ids.append(cube.attributes['tracking_id'])
+        except KeyError:
+            pass
+
+        to_remove = [
+            'time_coverage_start', 'time_coverage_end',
+            'history', 'tracking_id', 'start_date', 'stop_date',
+        ]
+        for attr in to_remove:
+            try:
+                del cube.attributes[attr]
+            except KeyError:
+                pass
+    return tracking_ids
+
+
+def _monthly_mean(cube):
```

Comment:
Isn't this now available as a preprocessor function?

Reply:
Good catch!